web/Views/Shared/_Layout.cshtml

```diff
@@ -86,11 +86,13 @@
 
 <main role="main">
     <div class="container mt-3 pb-4">
-        @if (TempData["FlashMessages"] is string[] flashMessages && TempData["FlashTypes"] is string[] flashTypes)
+        @if (TempData["FlashMessages"] != null && TempData["FlashTypes"] != null)
         {
-            for (var i = 0; i < flashMessages.Length; i++)
+            var flashMessages = TempData["FlashMessages"] as List<string>;
+            var flashTypes = TempData["FlashTypes"] as List<string>;
+            for (var i = 0; i < flashMessages!.Count; i++)
             {
-                <div class="alert alert-@flashTypes[i]" role="alert">
+                <div class="alert alert-@flashTypes![i]" role="alert">
                     @flashMessages[i]
                 </div>

```

Comment:
Why has this changed back?

Reply:
The way it was changed to, any new flash messages weren't displaying - the ones which were in the code previous to that change would still appear, but every time AddFlashMessage was added somewhere, the message would be added to TempData but would not be registered here by the as string[] and so wouldn't display.

Reply:
Hmm ok, sounds very very odd but ok